### ISSUE TITLE
Apptainer pushes image tags to the Image Repository of the GitLab Container Registry without created_at data

### ISSUE THREAD

**Issue by christianhueserhzdr** [apptainer/apptainer#2404]
### Version of Apptainer

What version of Apptainer (or Singularity) are you using? Run

`apptainer --version` (or `singularity --version`).

apptainer version 1.3.2

### Expected behavior

What did you expect to see when you do...?

The actual date and time of the image tags being pushed to the Image Repository of the GitLab Container Registry should be visible in the GitLab UI.

### Actual behavior

What actually happened? Why was it incorrect?

The date and time of the image tags being pushed to the Image Repository of the GitLab Container Registry is always "Published just now" in the GitLab UI regardless of the actual moment the image tag is pushed, i.e. even after days.

![gitlab-image-repository-tags-created-at-20240806](https://github.com/user-attachments/assets/b9d7814e-019a-4415-a6d7-2c36751127fc)

Apparently, the "created_at" data is missing for the image tag when accessing the information via the GitLab API:

```
{
  "name": "test",
  "path": "examples/apptainer:test",
  "location": "example.gitlab.com/examples/apptainer:test",
  "revision": "435243524563457rtez63534....",
  "short_revision": "435243524",
  "digest": "sha256:4e564356t54g5434t53t35...",
  "created_at": null,   <=====
  "total_size": 89440256
}
```

### Steps to reproduce this behavior

How can others reproduce this issue/problem?

* Push an image tag to the Image Repository of the GitLab Container Registry with Apptainer.

### What OS/distro are you running

Ubuntu 22.04

### How did you install Apptainer

Write here how you installed Apptainer (or Singularity). Eg. RPM, source.

```
add-apt-repository -y ppa:apptainer/ppa
apt-get update
apt install -qy apptainer
```

**Comment by DrDaveD** [comment]
I would think that the registry itself should be inserting that timestamp.

**Comment by DrDaveD** [comment]
We do get timestamps when publishing to the Github Container Registry.  See for example [released apptainer containers](https://github.com/apptainer/apptainer/pkgs/container/apptainer).

**Comment by cclerget** [comment]
Just an assumption, Gitlab might look at this field https://github.com/opencontainers/image-spec/blob/39ab2d54cfa8fe1bee1ff20001264986d92ab85a/specs-go/v1/config.go#L95, but the way apptainer push/store OCI artifacts on registries doesn't use `Image` (https://github.com/opencontainers/image-spec/blob/39ab2d54cfa8fe1bee1ff20001264986d92ab85a/specs-go/v1/config.go#L93) at all, that might explain why

**Comment by DrDaveD** [comment]
Oh, that makes me realize that probably the released apptainer containers aren't a good example because they aren't stored as OCI artifacts but actually as OCI/docker layers.  That is, apptainer has to read them with `docker://` and not `oras://`.

**Comment by afbjorklund** [comment]
> I would think that the registry itself should be inserting that timestamp.

No, they have be annotated when pushing. The `oras` tool is doing this...

`org.opencontainers.image.created`

You can assign this to me, it can be fixed while doing refactoring for #2896
